processor/src/decoder/mod.rs

```diff
@@ -268,6 +273,32 @@ impl Decoder {
         self.trace.program_hash()
     }
 
+    /// Returns an operation to be executed at the specified clock cycle. Only applicable in debug mode.
+    pub fn get_operation_at(&self, clk: usize) -> Operation {
+        self.operations[clk]
+    }
+
+    /// Returns the asmop decorator at the specified clock cycle.
+    /// - Returns asmop decorator if one exists at the specified clock cycle.
+    /// - Returns None otherwise
+    pub fn get_asmop_at(&self, asmop_idx: usize, clk: usize) -> Option<AsmOpInfo> {
+        if asmop_idx < self.asmop_list.len() && self.asmop_list[asmop_idx].0 == clk {
+            Some(self.asmop_list[asmop_idx].1.clone())
+        } else {
+            None
+        }
+    }
```

Comment:
I'm wondering whether we can simplify this structure a bit. Specifically, `asmop_list` is already sorted by clock cycle. So, given a clock cycle, we should be able to find an asmop at or right before it (we should be able to use binary search for this). Then, we can check (using `num_cycles` from asm op) whether the asm op is relevant for this specific cycle.

If this works, we won't need to pass `asmop_idx` as a separate parameter, which will make the interface much more easy to use.

Reply:
I did go the binary search route earlier but later changed it. Makes sense to keep the interface simpler. Will resolve this after we decide how to display AsmOpInfo in VmState.

Reply:
Another option could be to expose `asmop_list` (maybe a better name could be `assembly_ops`) via a getter (e.g., `pub fn assembly_ops(&self) -> &[(usize, AsmOpInfo)]`), and then determine which assembly op is relevant within the iterator.

Reply:
Makes sense. Made suggested changes.